pkg/iter/sample_iterator_test.go

```diff
@@ -170,13 +220,14 @@ func TestNewSampleQueryClientIterator(t *testing.T) {
 }
 
 func TestNewNonOverlappingSampleIterator(t *testing.T) {
+	// todo fix this test
```

Comment:
Can you please elaborate more in this comment for future reference?

Reply:
I guess I forgot about that I did fix it. :)

I add a lot of comment during the large refactoring to remember what to fix and that's a left over.